app/services/TransferAgreementService.scala

```diff
@@ -0,0 +1,30 @@
+package services
+
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken
+import configuration.GraphQLConfiguration
+import controllers.TransferAgreementData
+import graphql.codegen.AddTransferAgreement.AddTransferAgreement
+import graphql.codegen.types.AddTransferAgreementInput
+import services.ApiErrorHandling.sendApiRequest
+
+import java.util.UUID
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class TransferAgreementService @Inject()(val graphqlConfiguration: GraphQLConfiguration) (implicit val ec: ExecutionContext){
+  private val addTransferAgreementClient = graphqlConfiguration.getClient[AddTransferAgreement.Data, AddTransferAgreement.Variables]()
```

Comment:
Might be worth putting in a comment about Intellij showing an error at line 16, when in fact there isn't one. Just to avoid any confusion in the future.